### ISSUE TITLE
FindBugs plugin with `configure-on-demand` and `parallel` tries to read magic bytes of non-jar files

### ISSUE THREAD

**Issue by venantius** [gradle/gradle#1745]
This is essentially the same issue as was reported on the old bug forums: https://discuss.gradle.org/t/gradle-3-0-findbugstest-linux/18960

### Expected Behavior

The FindBugs plugin does not pick up non-jar files and try to treat them as jars when running in `configure-on-demand` and `parallel` (note that these flags were hypothesized to be related by @wolfs in the linked issue, above).

### Current Behavior

During a build, the FindBugs plugin picks up `.so` and `.dylib` files in the gradle cache and tries to treat them as jars. This then causes a failure due to those files not having zip magic bytes. 

A sample stacktrace (from our SQLite dep):

```
java.io.IOException: Wrong magic bytes of cafebabe for zip file /tmp/.gradle/caches/modules-2/files-2.1/com.almworks.sqlite4java/libsqlite4java-osx/1.0.392/de9528513a20ceae197ce9e14690973d617233bb/libsqlite4java-osx-1.0.392.dylib of 1620880 bytes
      At edu.umd.cs.findbugs.classfile.impl.ZipFileCodeBase.<init>(ZipFileCodeBase.java:87)
      At edu.umd.cs.findbugs.classfile.impl.ZipCodeBaseFactory.makeZipCodeBase(ZipCodeBaseFactory.java:39)
      At edu.umd.cs.findbugs.classfile.impl.ClassFactory.createFilesystemCodeBase(ClassFactory.java:121)
      At edu.umd.cs.findbugs.classfile.impl.FilesystemCodeBaseLocator.openCodeBase(FilesystemCodeBaseLocator.java:77)
      At edu.umd.cs.findbugs.classfile.impl.ClassPathBuilder.processWorkList(ClassPathBuilder.java:607)
      At edu.umd.cs.findbugs.classfile.impl.ClassPathBuilder.build(ClassPathBuilder.java:226)
      At edu.umd.cs.findbugs.FindBugs2.buildClassPath(FindBugs2.java:677)
      At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:218)
      At org.gradle.api.plugins.quality.internal.findbugs.FindBugsExecuter.runFindbugs(FindBugsExecuter.java:39)
```

### Context

This causes non-deterministic and infrequent build failures (but very annoying!). As our multi-project build continues to grow exponentially, this has started to happen much more frequently and is now quite disruptive.

**Comment by big-guy** [comment]
Is this with 3.0 or 3.4.1 or another version?

Which version of FindBugs are you using?

**Comment by venantius** [comment]
This is with 3.4.0 at the moment

**Comment by venantius** [comment]
I believe we just inherit the version of findbugs used by the Gradle plugin

**Comment by venantius** [comment]
The specific dependency that triggers this is sqlite4java, which is depended on by DynamoDBLocal

**Comment by blindpirate** [comment]
I believe this issue has nothing to do with `configure-on-demand` and `parallel`, and it seems not to fail the test (just pollutes the log). It's simply a problem of not excluding non-jar files when starting findbugs process. I have created #2167 to fix this.

**Comment by wolfs** [comment]
@venantius This should have been fixed by #2102 and is probably a duplicate of #1094. Can you confirm that your problem is fixed with the latest Gradle nightly?

**Comment by blindpirate** [comment]
Closed due to unresponsiveness.
